### PR TITLE
fix: Update GraphQL to return correct HTTP response code for file size limit upload errors

### PR DESCRIPTION
**Pull request by ardatan**
### New Pull Request Checklist

- [X ] I am not disclosing a [vulnerability](https://github.com/parse-community/parse-server/blob/master/SECURITY.md).
- [X] I am creating this PR in reference to an [issue](https://github.com/parse-community/parse-server/pull/8043#issuecomment-1155742360).

### Issue Description

Previously GraphQL Yoga was returning 500 but in the latest version it returns 400 which also not 100% correct. 
It should be 413 which indicates that user sends a request over the specified limit per HTTP spec.
https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/413

https://github.com/dotansimha/graphql-yoga/pull/1270
With the PR above on GraphQL Yoga repo, the behavior is fixed per spec. But Parse Server tests need to be updated since it was expecting the wrong status code (500) which indicates a misleading internal server error that isn't related to the user.

Related issue: https://github.com/parse-community/parse-server/pull/8043#issuecomment-1155742360

### TODOs before merging

- [ ] Add tests
- [ ] Add changes to documentation (guides, repository pages, in-code descriptions)
- [ ] Add [security check](https://github.com/parse-community/parse-server/blob/master/CONTRIBUTING.md#security-checks)
- [ ] Add new Parse Error codes to Parse JS SDK 
- [x] A changelog entry is created automatically using the pull request title (do not manually add a changelog entry)

### CODE REVIEW

**Comment by Moumouls** [comment]
Hi @ardatan, parse-server still use npm lockfile v1. So to install the project use `nvm use 14 && npm i` to avoid a lockfile migration. Then you should have less diffs.

**Comment by ardatan** [comment]
@Moumouls Sure I will try :)
Is there anything I can do so we can get CI running for this PR?

**Comment by mtrezza** [comment]
@ardatan if a PR is in the works and not ready for review yet you can change its status to "draft"; I've started the CI

**Comment by ardatan** [comment]
Could you run it again @mtrezza ?
Thanks!

**Comment by mtrezza** [comment]
Do you want to test anything specific in this CI that you cannot test locally by running `npm test`?

**Comment by ardatan** [comment]
We released a new version now, I updated the PR and ran tests locally it passes. 
Hopefully it will pass on CI as well. @mtrezza

**Comment by ardatan** [comment]
@mtrezza There was an issue with Node 12 support. So I updated the PR again.

**Comment by mtrezza** [comment]
Restarted CI

**Comment by Moumouls** [comment]
Hi @ardatan  thanks, i think we have merge conflicts here, related to the package.lock.

Could you rebase your PR on alpha and then push again (or fix package.lock conflicts) then we should be okay 🚀 

failing CI on node 17 is related to a flacky test on apple auth game center

**Comment by mtrezza** [comment]
I've resolved the conflicts, waiting for CI to pass

File: spec/ParseGraphQLServer.spec.js [hunk 1]
```diff
@@ -9419,7 +9419,8 @@ describe('ParseGraphQLServer', () => {
           });
 
           const result = JSON.parse(await res.text());
-          expect(res.status).toEqual(500);
+          // Server should return 413 https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/413
+          expect(res.status).toEqual(413);
```
Comment (mtrezza) [inline spec/ParseGraphQLServer.spec.js]:
```suggestion
          expect(res.status).toEqual(413);
```

**Comment by mtrezza** [comment]
Not sure how to handle this change of which is essentially a 3rd party-managed API. This is a clear example of why https://github.com/parse-community/parse-server/issues/7979 is so important to move GraphQL into an adapter, out of the core of Parse Server.

On the other hand if the file size limit is an undocumented GraphQL feature so far (or "experimental"), there may not even be a formal issue merging this as a "fix". As @n1ru4l [said](https://github.com/parse-community/parse-server/pull/8043#issuecomment-1165631783): "From our perspective, the file upload limit was undocumented and treated as unstable."

Suggested changelog entry for this PR:
> fix: update GraphQL to return correct HTTP response code for file size limit upload errors; the HTTP response code changes from `500` to `413`; the file size limit mechanism in GraphQL is an undocumented feature, if you are using this feature and parsing the response code make sure to adapt your app code accordingly before upgrading Parse Server.

@n1ru4l please correct me if the "is an undocumented feature" does not accurately describe it

**Comment by mtrezza** [comment]
What is the state of this PR?

**Comment by mtrezza** [comment]
Closing due to inactivity

**Comment by mtrezza** [comment]
@n1ru4l, @ardatan, @Moumouls Is this PR still needed? Or can we close it?

**Comment by Moumouls** [comment]
@mtrezza closing this since we are on apollo now

**Pull request closed without merging**